articles/house-price-prediction/index.md

```diff
@@ -0,0 +1,138 @@
+**ptratio** | *pupil-teacher ratio by town.*
+**black** | *1000(Bk - 0.63)^2 where Bk is the proportion of blacks by town.*
+**lstat** | *lower status of the population (percent).*
+**medv** | *median value of owner-occupied homes in \$1000s.*
+
+
+### Approach taken
+We'll use the Random Forest regression algorithm to predict the price of the houses. In this article, we'll consider machine learning algorithms as a black box that fits the data. 
+
+This article focuses more on the machine learning pipeline. For more information on the Random Forest algorithm, I suggest looking into this [video](https://www.youtube.com/watch?v=nxFG5xdpDto).
+
+We'll begin with loading the data. Since we are using an inbuilt dataset, we'll be calling the `load_boston` function from the `sklearn.datasets` module. The data is loaded into the `data` variable. Once the data is loaded, we separate the data and target attributes of the `data` variable. We store them in variables `data` and `target` respectively.
+
+Once we have the data and target values in 2 different variables, we can divide the data into two parts: the testing data and training data. The theory behind dividing the dataset into two parts is to ensure the model doesn't overfit the training data. Otherwise the model will perform well on the training data and perform poorly on the test data. 
+
+This means that the model has learned the training data so well that it cannot generalize new data points. We should avoid this. 
+
+Once we have the dataset split into training and testing sets, we can pre-process the data. Pre-processing involves scaling the values and converting the categorical values into numerical values. 
+
+For example, there is a variable in the given dataset that indicates whether the Charles river is close to the house or not. This variable takes the values `Near` and `Far.` 
+
+We need to convert this into a numerical value. To do this, we can use the `LabelEncoder` function available in the pre-processing module of sklearn. This will replace the column with numerical values of 0 and 1, respectively. 0 indicates `Near,` and 1 indicated `Far.`
+
+Once we perform the pre-processing of the dataset, we can fit the data to the model. We begin with instantiating an object of the `RandomForestRegressor` class. This is available in the `sklearn.ensemble` module. We use the `fit` method to fit the data to the model.
+
+Once the model is fit, we evaluate the model’s performance on the test set we got earlier. We use the `predict` method present in the `RandomForestRegressor` class. The `predict` method takes in the test input data and predicts an output. Using the predicted output and the actual output known from the dataset, we compute the test accuracy.
+
+Another useful evaluation metric is the `Mean Squared Error`. The MSE loss gives an estimate of how far the prediction is from the mean of the output. Computing the MSE gives us an idea about the performance of the algorithm.
```

Comment:
Consider adding the abbreviation next to the text to explicitly relate them:
`Mean Squared Error`(MSE)